dot/parachain/backing/second.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package backing
+
+import (
+	"errors"
+	"fmt"
+
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+var (
+	errWrongPVDForSecondingCandidate = errors.New(
+		"incorrect persisted validation data provided for seconding candidate")
+	errUnknownRelayParentForSecondingCandidate = errors.New(
+		"attempted to second a candidate with an unknown relay parent")
+	errParaOutsideAssignmentForSeconding = errors.New(
+		"subsystem requested to second for parachain beyond our assignment scope")
+	errAlreadySignedValidStatement = errors.New("already signed a valid statement for this candidate")
+)
+
+func (cb *CandidateBacking) handleSecondMessage(
+	candidateReceipt parachaintypes.CandidateReceipt,
+	pvd parachaintypes.PersistedValidationData,
+	pov parachaintypes.PoV,
+	chRelayParentAndCommand chan relayParentAndCommand,
+) error {
+	hash, err := candidateReceipt.Hash()
+	if err != nil {
+		return fmt.Errorf("hashing candidate receipt: %w", err)
+	}
+	candidateHash := parachaintypes.CandidateHash{Value: hash}
+
+	pvdBytes, err := scale.Marshal(pvd)
+	if err != nil {
+		return fmt.Errorf("marshalling persisted validation data: %w", err)
+	}
+
+	pvdHash, err := common.Blake2bHash(pvdBytes)
+	if err != nil {
+		return fmt.Errorf("hashing persisted validation data: %w", err)
+	}
+
+	if candidateReceipt.Descriptor.PersistedValidationDataHash != pvdHash {
+		return fmt.Errorf("%w: mismatch between %s and %s",
```

Comment:
This will just print two random hashes, which doesn't mean much to us. We could make this more informative like 
```suggestion
		return fmt.Errorf("%w: mismatch between persisted validation hash in candidate decriptor: %s and calculated hash of given persisted validation data: %s",
```